easybuild/tools/filetools.py

```diff
@@ -763,6 +763,18 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
+    if apatch.endswith(".gz"):
```

Comment:
This will only work if the patch if gzipped, what about `.zip`, `.bz2`, etc.?

Can we think of a better/more generic way to check whether we're dealing with a zipped file?

Also, there's a hard check in `fetch_patches` in `easyblock.py` on `.patch` extension, to distinguish between actual patch files and files that listed in `patches` are simply being copied, that will need to be enhanced too (maybe by changing the check to:

```python
if '.patch' not in patch_spec[0]:
```